pilot/cmd/pilot-agent/status/server.go

```diff
@@ -709,10 +709,13 @@ func (s *Server) handleAppProbeHTTPGet(w http.ResponseWriter, req *http.Request,
 	}
 
 	// Forward incoming headers to the application.
-	appReq.Host = req.Host
+	if prober.HTTPGet.Host != "" {
+		appReq.Host = prober.HTTPGet.Host
```

Comment:
better test with the depoyment to check why connection refused

Reply:
I replicated the health check failing with a 404 error. I am not sure why a connection refused error was seen.